addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -458,25 +211,17 @@ def _get_contextual_employee_id(self):
 
     @api.depends_context('employee_id', 'default_employee_id')
     def _compute_leaves(self):
-        data_days = {}
-        employee_id = self._get_contextual_employee_id()
-
-        if employee_id:
-            data_days = (self.get_employees_days(employee_id)[employee_id[0]] if isinstance(employee_id, list) else
-                         self.get_employees_days([employee_id])[employee_id])
-
+        employee_id = self.env['hr.employee'].browse(self._get_contextual_employee_id())
+        data_days = self.get_allocation_data(employee_id)[employee_id]
         for holiday_status in self:
-            result = data_days.get(holiday_status.id, {})
-            holiday_status.max_leaves = result.get('max_leaves', 0)
-            holiday_status.leaves_taken = result.get('leaves_taken', 0)
-            holiday_status.remaining_leaves = result.get('remaining_leaves', 0)
-            holiday_status.virtual_remaining_leaves = result.get('virtual_remaining_leaves', 0)
-            holiday_status.virtual_leaves_taken = result.get('virtual_leaves_taken', 0)
-            holiday_status.closest_allocation_to_expire = result.get('closest_allocation_to_expire', 0)
+            result = next((item for item in data_days if item[0] == holiday_status.name), ('', {}))
```

Comment:
Don't get why do you need `next` here :innocent: Why would you have ('', {}) in `data_days` :thinking:

Reply:
I used this to fetch a specific element in the list and get a default value if nothing comes out. The next method was set because the for operator returns an iterator if not set inside brackets, but it is indeed not that readable.